src/camera_capture/camera_capture/camera_capture.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""ROS2 CSI Camera Image Publisher.
+This script publishes csi camera image to a ROS2 topic in sensor_msgs.msg/Image format.
+Example:
+        $ colcon build --symlink-install
+        $ ros2 launch camera_snap_shot camera_snap_shot.launch.py
+"""
+
+# ___Import Modules:
+import os
+import cv2
+import json
+import numpy as np
+
+import rclpy
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+from ament_index_python.packages import get_package_share_directory
+from std_msgs.msg import Bool
+from cv_bridge import CvBridge
+
+from . import PACKAGE_NAME
+
+
+# ___Global Variables:
+SETTINGS = os.path.join(
+    get_package_share_directory("camera_capture"), "config/camera_capture_settings.json"
+)
+with open(SETTINGS) as fp:
+    json_settings = json.load(fp)
+
+
+# __Functions:
+def gstreamer_pipeline(
+    capture_width: str = str(json_settings["capture_width_livestream"]),
```

Comment:
A number of items in the json_settings are retrieved multiple times at multiple places. It may improve readability and clarity for readers to define constants for those settings e.g. `CAPTURE_WIDTH_LIVESTREAM = json_settings["capture_width_livestream"]`, and use `CAPTURE_WIDTH_LIVESTREAM` in all other parts of the code

Reply:
json_settings['image_location'] is another example

Reply:
Just occured to me: this also improves testability slightly